cclient/api/extended/prepare_transfers.c

```diff
@@ -5,27 +5,219 @@
  * Refer to the LICENSE file for licensing information
  */
 
-#include "cclient/api/extended/prepare_transfers.h"
+#include <math.h>
+#include <stdbool.h>
+
+#include "cclient/api/core/get_balances.h"
+#include "cclient/api/extended/get_inputs.h"
+#include "cclient/api/extended/get_new_address.h"
 #include "cclient/api/extended/logger.h"
+#include "cclient/api/extended/prepare_transfers.h"
+#include "utils/containers/hash/hash243_queue.h"
+#include "utils/time.h"
 
-retcode_t iota_client_prepare_transfers(iota_client_service_t const* const serv, transfer_t** const transfers,
-                                        uint32_t const num_transfer, bundle_transactions_t* out_bundle) {
-  retcode_t ret_code = RC_ERROR;
-  iota_transaction_t TX = {};
-  iota_transaction_t* tx = NULL;
-  transfer_iterator_t* transfer_iterator = NULL;
-  Kerl kerl = {};
+static void add_transaction_entry(bundle_transactions_t* bundle, int num_txs, flex_trit_t const* const address,
+                                  int64_t value, flex_trit_t* tag, uint64_t timestamp) {
+  iota_transaction_t tx = {};
+  for (int i = 0; i < num_txs; i++) {
+    transaction_reset(&tx);
```

Comment:
Only need to reset once before the loop since all the other fields will be changed every time